pallets/lbp/src/lib.rs

```diff
@@ -0,0 +1,987 @@
+					Error::<T>::NothingToUpdate
+				);
+
+				Self::ensure_pool_ownership(&who, &pool_id)?;
+
+				let now = <frame_system::Pallet<T>>::block_number();
+
+				ensure!(pool.start == Zero::zero() || now < pool.start, Error::<T>::SaleStarted);
+
+				pool.start = start.unwrap_or(pool.start);
+				pool.end = end.unwrap_or(pool.end);
+				if let Some(initial_weights) = initial_weights {
+					pool.initial_weights = Self::get_weights_in_order(pool, initial_weights)?;
+				}
+				if let Some(final_weights) = final_weights {
+					pool.final_weights = Self::get_weights_in_order(pool, final_weights)?;
+				}
+
+				Self::validate_pool_data(&pool)?;
+
+				Self::deposit_event(Event::PoolUpdated(who, pool_id));
+				Ok(().into())
+			})
+		}
+
+		#[pallet::weight(<T as Config>::WeightInfo::pause_pool())]
+		pub fn pause_pool(origin: OriginFor<T>, pool_id: PoolId<T>) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+
+			<PoolData<T>>::try_mutate_exists(pool_id.clone(), |maybe_pool| -> DispatchResultWithPostInfo {
+				// check existence of the pool
+				let mut pool = maybe_pool.as_mut().ok_or(Error::<T>::PoolNotFound)?;
+
+				Self::ensure_pool_ownership(&who, &pool_id)?;
+
+				ensure!(pool.pausable, Error::<T>::PoolIsNotPausable);
+				ensure!(!pool.paused, Error::<T>::CannotPausePausedPool);
+
+				let now = <frame_system::Pallet<T>>::block_number();
+				ensure!(pool.end > now, Error::<T>::CannotPauseEndedPool);
+
+				pool.paused = true;
+
+				Self::deposit_event(Event::Paused(who, pool_id));
+				Ok(().into())
+			})
+		}
+
+		#[pallet::weight(<T as Config>::WeightInfo::unpause_pool())]
+		pub fn unpause_pool(origin: OriginFor<T>, pool_id: PoolId<T>) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+
+			<PoolData<T>>::try_mutate_exists(pool_id.clone(), |maybe_pool| -> DispatchResultWithPostInfo {
+				// check existence of the pool
+				let mut pool = maybe_pool.as_mut().ok_or(Error::<T>::PoolNotFound)?;
+
+				Self::ensure_pool_ownership(&who, &pool_id)?;
+
+				ensure!(pool.paused, Error::<T>::PoolIsNotPaused);
+
+				let now = <frame_system::Pallet<T>>::block_number();
+				ensure!(pool.end > now, Error::<T>::CannotUnpauseEndedPool);
+
+				pool.paused = false;
+
+				Self::deposit_event(Event::Unpaused(who, pool_id));
+				Ok(().into())
+			})
+		}
+
+		#[pallet::weight(<T as Config>::WeightInfo::add_liquidity())]
+		#[transactional]
+		pub fn add_liquidity(
+			origin: OriginFor<T>,
+			pool_id: PoolId<T>,
+			amount_a: (AssetId, BalanceOf<T>),
+			amount_b: (AssetId, BalanceOf<T>),
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+
+			let (asset_a, asset_b) = (amount_a.0, amount_b.0);
+			let (amount_a, amount_b) = (amount_a.1, amount_b.1);
+
+			let asset_pair_id = T::AssetPairPoolId::from_assets(asset_a, asset_b);
+			ensure!(pool_id == asset_pair_id, Error::<T>::InvalidAsset);
```

Comment:
I dont see a point in constructing this pair id and performing the check.

Reply:
This check is used to ensure that we are not using this method to add wrong asset to the pool specified by pool_id param (e.g. ETH to HDX/ACA pool).

Reply:
It is probably not necessary to expect both the `pool_id` parameter and asset pair. `Pool_id` can be removed. What do you think?

Reply:
Well, it was not clear for me the purpose of this check. Now that i know - why not just to check PoolData's assets?

Reply:
we verify that the provided pool_id and assets target the same pool.

Reply:
`pool_id` param has been removed